frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -38,6 +40,64 @@
 from catalyst.utils.types import convert_pytype_to_shaped_array
 
 
+def accelerate(func=None, dev=None):
+    """Seamless integration with jax.jit
+
+    The accelerate function can be used to automatically apply `jax.jit` to
+    the input function `func` and be used to specify the device with the `dev`
+    keyword argument. If `dev` is None, then the default device as determined
+    by JAX is used.
+
+    Can be used in the following modes:
+
+
+    ```py
+    @accelerate
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    ```
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    or
+
+    ```
+    @qml.qjit
+    def qjitted_fn(x):
+      return accelerate(lambda x: x)
+    ```
+    """
+
+    if isinstance(func, Callable):
+
+        @functools.wraps(func)
+        def defer(*args, **kwargs):
+            # Make abstract variables from input tracers.
+            absargs, abskwargs = tree_map(shaped_abstractify, (args, kwargs))
+            # Find the shape of the return value
+            _, returnshape = jax.make_jaxpr(func, return_shape=True)(*absargs, **abskwargs)
+            jitted_fn = jax.jit(func, device=dev)
```

Comment:
@erick-xanadu I recommend maybe a `try`-`except` here. That way, if `jit` compilation fails, we can let the user know what went wrong. But we should still raise the original traceback as well.

E.g.,

```python
try:
    jitted_fn = jax.jit(func)
except Exception as e:
    raise ValueError("Error jitting func etc. etc.") from e
```

There may be something more specific you can use instead of catching all `Exceptions` here, not sure --- would require looking at `jax.jit`